builtin/providers/aws/resource_aws_cloudwatch_event_rule.go

```diff
@@ -247,3 +252,17 @@ func getStringStateFromBoolean(isEnabled bool) string {
 	}
 	return "DISABLED"
 }
+
+func validateEventPatternValue(length int) schema.SchemaValidateFunc {
+	return func(v interface{}, k string) (ws []string, errors []error) {
+		value := v.(string)
+		if len(value) > length {
+			errors = append(errors, fmt.Errorf(
+				"%q cannot be longer than %d characters: %q", k, length, value))
+		}
+		if _, err := normalizeJsonString(value); err != nil {
+			errors = append(errors, fmt.Errorf("%q contains an invalid JSON: %s", k, err))
+		}
```

Comment:
We send the **minified** (normalized) JSON to the API which is the one that is limited to 2048 characters, not the original JSON string with spaces, newlines etc.

At this point we would prevent the user from sending minified JSON to the API which can be below 2048 chars after stripping whitespaces etc. Also I'm not sure if there is any point in checking length of invalid JSON.

It may be worth covering this particular edge case by a test.